pkg/distributor/distributorerror/error.go

```diff
@@ -0,0 +1,121 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package distributorerror
+
+import (
+	"fmt"
+
+	"github.com/grafana/mimir/pkg/util/globalerror"
+	"github.com/grafana/mimir/pkg/util/validation"
+)
+
+type distributorPushError struct {
+	err error
+}
+
+// Error makes distributorPushError implement error interface.
+func (e distributorPushError) Error() string {
+	return e.err.Error()
+}
+
+// Unwrap makes distributorPushError implement Wrapper interface.
+func (e distributorPushError) Unwrap() error {
+	return e.err
+}
```

Comment:
In the initial version of this PR, `Unwrap` was needed to keep the semantics of the underlying error, so for example, we don't lose the info that the error shouldn't be logged. This is, in fact, not needed anymore, since we abandoned the idea of wrapping all the errors by the `DistributorPushError`. I will remove this.